dlib/image_transforms/interpolation.h

```diff
@@ -865,10 +865,18 @@ namespace dlib
                 float fout[4];
                 out.store(fout);
 
-                out_img[r][c]   = static_cast<T>(fout[0]);
-                out_img[r][c+1] = static_cast<T>(fout[1]);
-                out_img[r][c+2] = static_cast<T>(fout[2]);
-                out_img[r][c+3] = static_cast<T>(fout[3]);
+                const auto convert_to_output_type = [](float value)
+                {
+                    if (std::is_integral<T>::value)
+                        return static_cast<T>(value + 0.5);
```

Comment:
use `std::round()` instead of adding 0.5

Reply:
Actually, I would rather not. In my experience `std::round()` can be much slower, at least on my machine (compiler). And in this kind of use it may really make a difference, I think.

Why is it slower? Guessing because of error handling, probably similar to `std::floor()` as discussed here: https://stackoverflow.com/a/824233/19254.

Reply:
Huh, that’s too bad. Yeah. Don’t use round then.